helm/oauth2-proxy/templates/deployment.yaml

```diff
@@ -112,11 +112,17 @@ spec:
         {{- if eq (default "cookie" .Values.sessionStorage.type) "redis" }}
         - name: OAUTH2_PROXY_SESSION_STORE_TYPE
           value: "redis"
-        {{- if or .Values.sessionStorage.redis.password .Values.sessionStorage.redis.existingSecret }}
+         {{- if or .Values.sessionStorage.redis.existingSecret .Values.sessionStorage.redis.password (and .Values.redis.enabled (.Values.redis.auth).enabled )}}
         - name: OAUTH2_PROXY_REDIS_PASSWORD
           valueFrom:
             secretKeyRef:
-              name: {{ if .Values.sessionStorage.redis.existingSecret }} {{ .Values.sessionStorage.redis.existingSecret }}{{ else }} {{ template "oauth2-proxy.fullname" . }}-redis-access{{ end }}
+              {{- if .Values.sessionStorage.redis.existingSecret }}
```

Comment:
Shouldn't we do the same for sentinel? Note that even without it, this is still an improvement!

Reply:
I guess yes, I didn't understand how this with sentinel works TBH. But I haven't really looked into that. I can take a look later and see if I can figure it out.